internal/namespace/validator.go

```diff
@@ -108,6 +119,39 @@ func (v *Validator) handle(req *nsRequest) admission.Response {
 	return webhooks.Allow("")
 }
 
+// illegalTreeLabel checks if tree labels are being created or modified
+// by any user or service account since only HNC service account is
+// allowed to do so
+func (v *Validator) illegalTreeLabel(req *nsRequest) admission.Response {
+	msg := "Cannot set or modify tree label %q in namespace %q; these can only be modified by HNC."
+
+	oldLabels := map[string]string{}
+	if req.oldns != nil {
+		oldLabels = req.oldns.Labels
+	}
+	// Ensure the users hasn't added or changed any tree labels
+	for key, val := range req.ns.Labels {
+		if !strings.Contains(key, api.LabelTreeDepthSuffix) {
+			continue
+		}
+
+		// Check if new HNC label tree key isn't being added
+		if oldVal, ok := oldLabels[key]; !ok || val != oldVal {
```

Comment:
I think you can simplify this to:
```
if oldLabels[key] != val
```
If the key isn't in oldLabels, `oldLabels[key]` will evaluate to the empty string so this case will work for adds as well.